code/src/wattrex_battery_cycler/mid/mid_data/mid_data_experiment.py

```diff
@@ -137,11 +116,43 @@ def __init__(self, volt_max : int|None = None, volt_min : int|None = None,
             curr_max (int): [description]
             curr_min (int): [description]
         '''
+        if ((volt_max is None and #pylint: disable=too-many-boolean-expressions
+             (volt_min is not None or curr_max is not None or curr_min is not None))
```

Comment:
Much easier, if any is none, then raise an error

Reply:
Makes sense, done